integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java

```diff
@@ -0,0 +1,184 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.esb.mediator.test.log;
+
+import org.apache.axiom.om.OMElement;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.clients.logging.LoggingAdminClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+/**
+ * THis test case verifies different levels and categories in log mediator
+ */
+public class LogMediatorLevelsAndCategoryTestCase extends ESBIntegrationTest {
+
+    private LogViewerClient logViewer;
+    private LoggingAdminClient logAdmin;
+    private OMElement response;
+    private LogEvent[] logs;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        logAdmin = new LoggingAdminClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer.clearLogs();
+        verifyProxyServiceExistence("LogMediatorLevelAndCategoryTestProxy");
+        logAdmin.updateLoggerData("org.apache.synapse.mediators.builtin",
+                LoggingAdminClient.LogLevel.TRACE.name(),true, false);
+        response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("LogMediatorLevelAndCategoryTestProxy"),
+                        null, "WSO2");
+        Assert.assertTrue(response.toString().contains("WSO2"));
+        logs = logViewer.getAllRemoteSystemLogs();
+    }
+
+    @Test(groups = "wso2.esb", description = "Test debug level log")
+    public void testDebugLevelLogs() throws Exception {
+        Boolean isDebugLogAvailable = false;
+        //LogEvent[] getLogsDebug = logViewer.getLogs("DEBUG", "LogMediator", "", "");
```

Comment:
Can we remove this if this line is not needed?

Reply:
Sure, Will do